### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

**Comment by massich** [comment]
[Stackoverflow notes on SWIG and doxygen docstrings](https://stackoverflow.com/questions/37780325/generating-doxygen-comments-for-swig-generated-c-sharp-that-wraps-c)

**Comment by massich** [comment]
the `doxy2siwg` command needs to be remade! 

the docstrings have never been used. The generated file `docstings.i` to import from **swig** is appended here `SWIG_MODULE_OPENMEEG_EXTRA_DEPS` and this varaible never used. (see this 5a4ef47a)


the entire wrapping needs a review. There are things like this variable `CMAKE_SWIG_FLAGS` that drive the generation process which was meant to be used by `UseSwig_AlwaysShared.cmake` but was removed in abcc0a25 and the variables never used.

**Comment by massich** [comment]
`SWIG_MODULE_${MODULE_NAME}_EXTRA_DEPS` is an internal variable for `siwg_add_module`. So `SWIG_MODULE_XXXX_EXTRA_DEPS`should fit with `swig_add_module(XXXX python regular_file.i)`

**Comment by massich** [comment]
I rebase master.

**Comment by massich** [comment]
[fix BUILD_DOCUMENTATION with Python bindings cf #278](https://trello.com/c/iey753v9/101-fix-builddocumentation-with-python-bindings-cf-278)

**Comment by agramfort** [comment]
merge when you're happy (enough)

thx @massich

File: documentation/reference/CMakeLists.txt [hunk 1]
```diff
@@ -9,10 +9,22 @@ IF (BUILD_DOCUMENTATION)
...
+    #       The way to do that is:
+    #       find_package(Doxygen
+    #                    REQUIRED dot
+    #                    OPTIONAL_COMPONENTS gnuplot)
+    #       if(NOT DOXYGEN_DOT_FOUND)
+    #            message(WARNING "Graphviz doesn't seem to be installed.\
+    #                             Doxygen will not be able to generate graphs.\
+    #                             Consider installing this package.")
+    #       endif(NOT DOXYGEN_DOT_FOUND)
+    #
```
Comment (agramfort) [inline documentation/reference/CMakeLists.txt]: shall we clean this?

File: documentation/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,15 @@
-IF ( BUILD_DOCUMENTATION )
+# # Require dot, treat the other components as optional
+# find_package(Doxygen
+#              REQUIRED dot) # XXX if dot module if not present, cmake should stop generatio
+#              # OPTIONAL_COMPONENTS mscgen dia)
```
Comment (agramfort) [inline documentation/CMakeLists.txt]: here also?

File: documentation/reference/CMakeLists.txt (hunk 1 above)
Comment (massich) [reply documentation/reference/CMakeLists.txt]: No. This is here to point how this should be done. Either we change it and do it properly or add XXX comment, move on and let someone else pick it up (or Sunday afternoon).

Comment (massich) [reply documentation/reference/CMakeLists.txt]: I mean

Comment (massich) [reply documentation/reference/CMakeLists.txt]: + it should be done at least one layer above. (in your other [comment](https://github.com/openmeeg/openmeeg/pull/289/files/a5ccaffde385aecc7c909b170c9e2fba4933c54a#r178302874) documentation/CMakeLists.txt)

Comment (massich) [reply documentation/reference/CMakeLists.txt]: But if we end up moving to fully documenting with Sphinx without doxygen, there would be no need for fixing it.

**Pull request merged**